### PR TITLE
feat(task): remove DefaultTaskResult

### PR DESCRIPTION
**Pull request by LioRoger**
#### What type of PR is this?

type-feature
#### What this PR does / why we need it:
1. remove default task result, cause task result is inner concept, no interface needed.
2. fix JobEntity.resultJson not set correct value leads to Json deserialize failure
#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-common/src/main/java/com/oceanbase/odc/common/util/MapUtils.java [hunk 1]
```diff
@@ -128,4 +129,39 @@ public static int size(final Map<?, ?> map) {
         return org.apache.commons.collections4.MapUtils.size(map);
     }
 
+
+    private int getMapSize(Map<String, String> map) {
```
Comment (yizhouxw) [inline server/odc-common/src/main/java/com/oceanbase/odc/common/util/MapUtils.java:133]: already has method `public static int size(final Map<?, ?> map) `

File: server/odc-common/src/main/java/com/oceanbase/odc/common/util/MapUtils.java [hunk 2]
```diff
@@ -128,4 +129,39 @@ public static int size(final Map<?, ?> map) {
...
+     * compare map, null map and empty map consider as equals
+     * 
+     * @param src
+     * @param target
+     * @param <Key> key of map
+     * @param <Value> value of map
+     * @param equalFunction function with argument value to compare if value instance is equals
+     * @return true is map is equals
+     */
+    public static <Key, Value> boolean mapEquals(Map<Key, Value> src, Map<Key, Value> target,
```
Comment (yizhouxw) [inline server/odc-common/src/main/java/com/oceanbase/odc/common/util/MapUtils.java:147]: may naming as `isEqual` , as currrent class name 'MapUtils' already has 'Map' concept

File: server/odc-common/src/test/java/com/oceanbase/odc/common/util/MapUtilsTest.java [hunk 3]
```diff
@@ -97,4 +98,37 @@ public void formatKvString_ValueContainsSeparator_IllegalArgumentException() {
         map.put("a", "2,");
         MapUtils.formatKvString(map);
     }
+
+    @Test
+    public void mapEqualsTest() {
```
Comment (yizhouxw) [inline server/odc-common/src/test/java/com/oceanbase/odc/common/util/MapUtilsTest.java:103]: unit testing case naming follow style:
`{methodUnderTest}_{scenario}_{expectedResult}`

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/TaskResult.java [hunk 4]
```diff
@@ -17,27 +17,48 @@
...
+    private String errorMessage;
 
-    String getErrorMessage();
+    private double progress;
 
-    double getProgress();
+    private Map<String, String> logMetadata;
 
-    Map<String, String> getLogMetadata();
+    public boolean progressChanged(TaskResult previous) {
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/TaskResult.java:49]: progressChanged --> isProgressChanged

File: server/odc-common/src/main/java/com/oceanbase/odc/common/util/MapUtils.java (hunk 1 above)
Comment (LioRoger) [reply server/odc-common/src/main/java/com/oceanbase/odc/common/util/MapUtils.java:133]: unused method， has removed

File: server/odc-common/src/main/java/com/oceanbase/odc/common/util/MapUtils.java (hunk 2 above)
Comment (LioRoger) [reply server/odc-common/src/main/java/com/oceanbase/odc/common/util/MapUtils.java:147]: agree

File: server/odc-common/src/test/java/com/oceanbase/odc/common/util/MapUtilsTest.java (hunk 3 above)
Comment (LioRoger) [reply server/odc-common/src/test/java/com/oceanbase/odc/common/util/MapUtilsTest.java:103]: split in four methods

**Review by yizhouxw — APPROVED** [review]
lgtm

File: server/odc-common/src/main/java/com/oceanbase/odc/common/util/MapUtils.java [hunk 5]
```diff
@@ -128,4 +129,34 @@ public static int size(final Map<?, ?> map) {
...
+     * compare map, null map and empty map consider as equals
+     * 
+     * @param src
+     * @param target
+     * @param <Key> key of map
+     * @param <Value> value of map
+     * @param equalFunction function with argument value to compare if value instance is equals
+     * @return true is map is equals
+     */
+    public static <Key, Value> boolean isEqual(Map<Key, Value> src, Map<Key, Value> target,
```
Comment (yhilmare) [inline server/odc-common/src/main/java/com/oceanbase/odc/common/util/MapUtils.java:142]: `equals` may be better?
just like `Objects.equals()`

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**